File: tests/fixtures/sample_files/java/Sample.java
/**
 * Sample Java file for parser testing.
 *
 * Expected values (hand-verified):
 *   Classes:    2  (Calculator, MathUtils)
 *   Methods:    4  (Calculator: constructor, add, reset; MathUtils: factorial)
 *   Imports:    java.util
 *   Interfaces: 0
 *   CC(add)     = 3  (if / else if / else)
 *   CC(factorial) = 2  (if)
 */

import java.util.ArrayList;
import java.util.List;

public class Calculator {
    private int value;
    private List<Integer> history;

    public Calculator(int initial) {
        this.value = initial;
        this.history = new ArrayList<>();
    }

    public Calculator add(int x) {
        if (x > 0) {
            this.value += x;
        } else if (x < 0) {
            this.value += x;
        } else {
            // no-op
        }
        this.history.add(x);
        return this;
    }

    public void reset() {
        this.value = 0;
        this.history.clear();
    }
}

class MathUtils {
    public static int factorial(int n) {
        if (n <= 1) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}

File: tests/fixtures/sample_files/typescript/sample.ts
/**
 * Sample TypeScript file for parser testing.
 *
 * Expected values (hand-verified):
 *   Classes:    1  (Calculator)
 *   Interfaces: 1  (Computable)
 *   Methods:    3  (constructor, add, reset)
 *   Functions:  1  (helper)
 *   Imports:    path
 *   Enums:      1  (Operation)
 */

import * as path from "path";

interface Computable {
    add(x: number): Computable;
    reset(): void;
}

enum Operation {
    Add = "add",
    Reset = "reset",
}

class Calculator implements Computable {
    private value: number;
    private history: number[];

    constructor(initial: number = 0) {
        this.value = initial;
        this.history = [];
    }

    add(x: number): Calculator {
        if (x > 0) {
            this.value += x;
        } else if (x < 0) {
            this.value += x;
        } else {
            // no-op
        }
        this.history.push(x);
        return this;
    }

    reset(): void {
        this.value = 0;
        this.history = [];
    }
}

function helper(a: number, b: number, c: number = 0): number {
    return a + b + c;
}

export { Calculator, helper, Computable, Operation };
